File: server/piwars-common/src/main/java/org/ah/piwars/virtualrover/game/rovers/CBISRover.java
package org.ah.piwars.virtualrover.game.rovers;

import com.badlogic.gdx.math.Quaternion;
import com.badlogic.gdx.math.Vector2;
import com.badlogic.gdx.math.Vector3;

import org.ah.themvsus.engine.common.game.GameObjectFactory;

import static org.ah.piwars.virtualrover.engine.utils.CollisionUtils.asShape2DList;
import static org.ah.piwars.virtualrover.engine.utils.CollisionUtils.polygonFromBox;

public class CBISRover extends Rover {

    public CBISRover(GameObjectFactory factory, int id) {
        super(factory, id, RoverType.CBIS);

        this.polygons = asShape2DList(polygonFromBox(-100f,  -75f, 100f,  75f));
        this.attachmentPosition = new Vector2(100f, 0);
        this.cameraPosition = new Vector3(100f, 0f, 30f);
        this.cameraOrientation = new Quaternion();
        this.cameraAngle = 45f;
    }
}

File: fishtank/server/fishtank-python/src/main/python/fishtankclient/engine/simulation/rovers/CBISRover.py

from enum import Enum

from fishtankclient.engine.simulation.rovers.AbstractRoverSimObject import AbstractRoverSimObject
from fishtankclient.engine.simulation.rovers.RoverType import RoverType
from fishtankclient.engine.utils import Polygon


class CBISRover(AbstractRoverSimObject):

    def __init__(self, factory, sim_object_id, sim_object_type):
        super(CBISRover, self).__init__(factory, sim_object_id, sim_object_type, RoverType.CBISRover)
        self.attachment_position = (100.0, 0.0)
        self.camera_position = (100.0, 0.0, 15.0)
        self.camera_orientation = [0.0, 0.0, 0.0, 1.0]
        self.camera_angle = 45.0

    def __repr__(self):
        return "CBISRover[" + super(CBISRover, self).__repr__() + "]"

    def get_shape(self):
        return Polygon.box(200, 150)